src/pymoca/backends/casadi/model.py

```diff
@@ -728,22 +728,35 @@ def get_derivative(expr):
             # Operations that preserve the equivalence of an equation
             # TODO: There may be more, but this is the most frequent set
             unary_ops = ca.OP_NEG, ca.OP_FABS, ca.OP_SQRT
-            binary_ops = ca.OP_MUL, ca.OP_DIV
 
             # Recursive factor and simplify function
             def factor_and_simplify(eq):
                 # These are ops that can simply be dropped
                 if eq.n_dep() == 1 and eq.op() in unary_ops:
                     return factor_and_simplify(eq.dep())
 
-                # These are binary ops and can get a little tricky
-                # For now, we just drop constant divisors or multipliers
-                elif eq.n_dep() == 2 and eq.op() in binary_ops:
-                    if eq.dep(1).is_constant():
-                        return factor_and_simplify(eq.dep(0))
-                    elif eq.dep(0).is_constant() and eq.op() == ca.OP_MUL:
-                        return factor_and_simplify(eq.dep(1))
-
+                elif eq.n_dep() == 2:
+                    # Drop constant divisors or multipliers
```

Comment:
Not your change, but I'm thinking this simplification is actually rather dangerous. For scaling of equations we actually might want to allow dividing an equation by a constant... I always feel rather unsure about solely leaving it to a solver like IPOPT to scale constraints.

Reply:
Well, the elimination of constant dividers was there before (together with constant multipliers). I am only adding the additions and subtractions binary operations that can be merged with the negative unary operations. 

The point you raise about IPOPT is definitely true. In any case, I am not sure that scaling a variable directly in the Modelica model would be the best way to go. If you were to tweak the Modelica model with the solution process already in mind (be it an optimization or a simple simulation) then you could simply leave the simplification options deactivated.